redfish-core/lib/pcie.hpp

```diff
@@ -67,8 +67,36 @@ static inline void
         {
             for (const auto& [serviceName, interfaceList] : serviceMap)
             {
-                std::string devName =
-                    sdbusplus::message::object_path(pcieDevicePath).filename();
+                sdbusplus::message::object_path path(pcieDevicePath);
+                sdbusplus::message::object_path parentPath = path.parent_path();
+                sdbusplus::message::object_path parentofParentPath =
+                    parentPath.parent_path();
+                std::string devName;
+
+                /**
+                 * @brief Workaround to handle duplicate PCI device list
+                 *
+                 * retrieve PCI device endpoint information and if path is
+                 * ~/chassisN/io_moduleN/slotN/adapterN then, replace redfish
+                 * PCI device as "chassisN_io_moduleN_slotN_adapterN"
+                 *
+                 */
+
+                if (parentofParentPath.parent_path().filename().find(
+                        "chassis") != std::string::npos)
```

Comment:
Should we use "chassis" here? I think we will have a Mex Drawerer chassis and maybe it will also have pcie devices idk?

Reply:
I don't think we should hardcode any chassis id here.. Why can't we just do this for all pcie devices? 
Can we just add the parent and the grandparent to all pcie devices? (or whatever the ancestors are we are adding?)

Reply:
Do you mean all parent info for all PCI devices?

Reply:
remove check is that chassis or not